pkgs/by-name/se/serial-studio/package.nix

```diff
@@ -2,34 +2,44 @@
   lib,
   stdenv,
   fetchFromGitHub,
-  qmake,
-  qtquickcontrols2,
-  qtserialport,
-  qtsvg,
-  wrapQtAppsHook,
+  cmake,
+  qt6,
+  pkg-config,
 }:
 
 stdenv.mkDerivation rec {
   pname = "serial-studio";
-  version = "1.1.7";
+  version = "3.0.6";
 
   src = fetchFromGitHub {
     owner = "Serial-Studio";
     repo = "Serial-Studio";
     rev = "v${version}";
-    hash = "sha256-Tsd1PGB7cO8h3HDifOtB8jsnj+fS4a/o5nfLoohVLM4=";
+    hash = "sha256-q3RWy3HRs5NG0skFb7PSv8jK5pI5rtbccP8j38l8kjM=";
     fetchSubmodules = true;
   };
 
   nativeBuildInputs = [
-    qmake
-    wrapQtAppsHook
+    cmake
+    qt6.wrapQtAppsHook
+    pkg-config
   ];
 
   buildInputs = [
-    qtquickcontrols2
-    qtserialport
-    qtsvg
+    qt6.qtbase
+    qt6.qtdeclarative
+    qt6.qtsvg
+    qt6.qtgraphs
+    qt6.qtlocation
+    qt6.qtconnectivity
+    qt6.qttools
+    qt6.qtserialport
+    qt6.qtpositioning
+  ];
+
+  patches = [
+    ./0001-CMake-Deploy-Fix.patch
```

Comment:
Would be cool if we could upstream those patches for the next release.

Reply:
I submitted an issue to upstream regarding the second patch (Connect Button Fix) since it was also present in the upstream provided AppImage.

I am not planning on telling upstream how to structure their CMake based packaging procedures. I did a quick fix that threw a lot of stuff out that caused issues with the Nix way of handling CMake. But I don't think that this would be suitable for upstream and I don't know a good way to harmonize the Nix and upstream requirements, mainly because I am not aware of the latter.

Reply:
This shouldn't be nix specific though. Maybe the qt_generate_deploy_qml_app_script, I am not sure about that but the other changes are just fixes that you can install the files to a custom prefix. I am not sure if the cmake docs has proper examples for that.